src/components/device-screen.tsx

```diff
@@ -11,84 +11,242 @@ import { Devices } from '../lib/xively/models/index';
+import xively from '../lib/xively';
+import { TimeSeries } from '../lib/xively/models/timeseries';
+import { groupBy } from 'lodash';
+import randomColor from 'randomcolor';
+import moment from 'moment';
 
 interface ReduxStateProps {
-    device: DeviceWithData;
+  device: DeviceWithData;
 }
 interface ReduxDispatchProps {
 }
 interface DeviceProps extends
-    React.Props<DeviceScreenComponent>,
-    ReduxStateProps,
-    ReduxDispatchProps,
-    NavigationScreenConfigProps {
+  React.Props<DeviceScreenComponent>,
+  ReduxStateProps,
+  ReduxDispatchProps,
+  NavigationScreenConfigProps {
 }
 
 function mapStateToProps(state: AppState, ownProps: DeviceProps) {
-    const device: Devices.Device = getDevice(state, ownProps.navigation.state.params.deviceId);
-    let mqttData: { [topicName: string]: TopicData } = {};
-    Object.keys(state.mqtt.data).filter((topic) => {
-        return topic.split('/')[5] === device.id;
-    }).forEach((topic) => {
-        mqttData[topic.split('/').slice(6).join('/')] = state.mqtt.data[topic];
-    });
-    return {
-        device:
-        { device, mqttData }
-    };
+  const device: Devices.Device = getDevice(state, ownProps.navigation.state.params.deviceId);
+
+  let mqttData: { [topicName: string]: TopicData } = {};
+
+  Object.keys(state.mqtt.data).filter((topic) => {
+    return topic.split('/')[5] === device.id;
+  }).forEach((topic) => {
+    mqttData[topic.split('/').slice(6).join('/')] = state.mqtt.data[topic];
+  });
+
+  return {
+    device:
+    { device, mqttData }
+  };
 }
 
 function mapDispatchToProps(dispatch: Dispatch<AppState>, ownProps: DeviceProps) {
-    return {
-    }
+  return {
+  }
 }
 
 interface DeviceState {
+  timeSeries?: Array<TimeSeries.DataPoint> | 'error';
 }
 
 export class DeviceScreenComponent extends React.Component<DeviceProps, DeviceState> {
-    connectedImage = require('../../images/device_on.png');
-    disconnectedImage = require('../../images/device_off.png');
-    static navigationOptions = {
-        title: 'Device',
-    };
-
-    render() {
-        if (!this.props || !this.props.device || !this.props.device.device) {
-            return (
-                <View style={Styles.container}>
-                    <Text style={Styles.title}>
-                        loading
-                    </Text>
-                </View>
-            )
+  connectedImage = require('../../images/device_on.png');
+  disconnectedImage = require('../../images/device_off.png');
+
+  static navigationOptions = {
+    title: 'Device',
+  };
+
+  state: DeviceState = {
+    timeSeries: null,
+  };
+
+  componentDidMount() {
+    return this.loadTimeSeries();
+  }
+
+  async loadTimeSeries() {
+    try {
+      this.setState({ timeSeries: null });
+
+      const { result } = await xively.timeseries.getLatestValues({
+        deviceId: this.props.device.device.id,
+        channel: 'charge-controller',
```

Comment:
I forget if there is anything hard-coded in RIoT like this, but if not please parameterize similarly to how we do on web.

Reply:
yes, I took it from there
https://github.com/ionia-corporation/xively-react-boilerplate/blob/dev/src/client/components/device/device-report.tsx#L32